src/common/interactive_retriever.py

```diff
@@ -0,0 +1,214 @@
+"""
+Retrieves candidates for a given set of queries after embedding them.
+"""
+
+from enum import Enum
+import gc
+import json
+import os
+
+import numpy as np
+import torch
+import torch.distributed as dist
+from torch.utils.data import DataLoader
+from torch.nn.parallel import DistributedDataParallel as DDP
+
+from data.mbeir_dataset import (
+    MBEIRInferenceOnlyDataset,
+    MBEIRInferenceOnlyCollator,
+)
+import dist_utils
+from dist_utils import ContiguousDistributedSampler
+from mbeir_embedder import generate_embeds_and_ids_for_dataset_with_gather
+from utils import build_model_from_config, set_seed
+from data.preprocessing.utils import unhash_did
+
+
+class QueryModality(Enum):
+    TEXT = "text"
+    IMAGE = "image"
+
+
+class InteractiveRetriever:
```

Comment:
I noticed that the InteractiveRetriever requires a pre-built candidate index file to function correctly. To assist users with this setup, could we consider adding a script, such as `run_interactive_retriever_pipeline.sh`, that demonstrates the entire pipeline? This script would cover embedding, indexing, and loading the index for the interactive retriever and retrieve demo queries. Additionally, incorporating a step-by-step guide in the README could greatly enhance the user experience.